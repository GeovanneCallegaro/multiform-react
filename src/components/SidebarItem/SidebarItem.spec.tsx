import { render, screen } from "@testing-library/react"
import { SidebarItem } from "."
import {BrowserRouter} from 'react-router-dom'

describe('<SidebarItem />', () => {
  it('should be on screen', () => {
    render(
      <BrowserRouter>
        <SidebarItem
          title = 'title1'
          description = "body1"
          icon = "profile"
          path = '/'
          active = {false}
        />
      </BrowserRouter>)

    expect(screen.getByText('title1')).toBeInTheDocument();
    expect(screen.getByText('body1')).toBeInTheDocument();
    expect(screen.getByText('profile.svg')).toBeInTheDocument();
    expect(screen.queryByText('book.svg')).not.toBeInTheDocument();
  })

  it('should render an other svg from icon atributte is book', () => {
    render(
      <BrowserRouter>
        <SidebarItem
          title = 'title1'
          description = "body1"
          icon = "book"
          path = '/'
          active = {false}
        />
      </BrowserRouter>)

    expect(screen.getByText('book.svg')).toBeInTheDocument();
  })

  it('should render an other svg from icon atributte is mail', () => {
    render(
      <BrowserRouter>
        <SidebarItem
          title = 'title1'
          description = "body1"
          icon = "mail"
          path = '/'
          active = {false}
        />
      </BrowserRouter>)

    expect(screen.getByText('mail.svg')).toBeInTheDocument();
  })

  it('should render an other svg from icon atributte is check', () => {
    render(
      <BrowserRouter>
        <SidebarItem
          title = 'title1'
          description = "body1"
          icon = "check"
          path = '/'
          active = {false}
        />
      </BrowserRouter>)

    expect(screen.getByText('check.svg')).toBeInTheDocument();
  })

  it('should match snapshot', () => {
    const {container} = render(
      <BrowserRouter>
        <SidebarItem
          title = 'title1'
          description = "body1"
          icon = "check"
          path = '/'
          active = {false}
        />
      </BrowserRouter>)

    expect(container.firstElementChild).toMatchSnapshot()
  })
})