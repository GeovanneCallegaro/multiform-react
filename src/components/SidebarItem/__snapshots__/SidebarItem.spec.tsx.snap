// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SidebarItem /> should match snapshot 1`] = `
<div
  class="sc-bczRLJ nvUja"
>
  <a
    href="/"
  >
    <div
      class="sc-gsnTZi kgUNVx"
    >
      <div
        class="sc-dkzDqf eiLjQC"
      >
        title1
      </div>
      <div
        class="sc-hKMtZM gnrDwn"
      >
        body1
      </div>
    </div>
    <div
      class="sc-eCYdqJ csAXp"
    >
      <svg
        fill="white"
        height="24"
        width="24"
      >
        check.svg
      </svg>
    </div>
    <div
      class="sc-jSMfEi etUSDR"
    />
  </a>
</div>
`;
